### PR TITLE
MAILBOX-272 Add users in ES indexer

### PR DESCRIPTION
**Pull request by aduprat**
Only last commit.

### CODE REVIEW

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java [hunk 1]
```diff
@@ -28,24 +28,27 @@
...
+import com.github.steveash.guavate.Guavate;
 import com.google.common.base.Preconditions;
 import com.google.common.base.Throwables;
 import com.google.common.collect.Multimap;
 
 public class IndexableMessage {
 
-    public static IndexableMessage from(MailboxMessage message, TextExtractor textExtractor, ZoneId zoneId) {
+    public static IndexableMessage from(MailboxMessage message, List<User> users, TextExtractor textExtractor, ZoneId zoneId) {
         Preconditions.checkNotNull(message.getMailboxId());
```
Comment (mbaechler) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java]: `Preconditions.checkArgument(!users.isEmpty());` ?

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/query/QueryConverter.java [hunk 2]
```diff
@@ -61,6 +63,17 @@ private QueryBuilder generateQueryBuilder(SearchQuery searchQuery) {
         }
     }
 
+    private QueryBuilder addUsersFilter(QueryBuilder queryBuilder, List<User> users) {
+        if (users.isEmpty()) {
```
Comment (mbaechler) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/query/QueryConverter.java]: should never happen ?

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMailboxMessageSearchIndexTest.java [hunk 3]
```diff
@@ -70,17 +70,20 @@ public void setup() throws JsonProcessingException {
 
         indexer = control.createMock(ElasticSearchIndexer.class);
 
-        expect(messageToElasticSearchJson.convertToJson(anyObject(MailboxMessage.class))).andReturn("json content").anyTimes();
+        expect(messageToElasticSearchJson.convertToJson(anyObject(MailboxMessage.class), anyObject())).andReturn("json content").anyTimes();
```
Comment (mbaechler) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMailboxMessageSearchIndexTest.java]: what is anyObject ?

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/MailboxMessageToElasticSearchJsonTest.java [hunk 4]
```diff
@@ -81,7 +83,7 @@ public void spamEmailShouldBeWellConvertedToJson() throws IOException {
             propertyBuilder,
             MAILBOX_ID);
         spamMail.setModSeq(MOD_SEQ);
-        assertThatJson(messageToElasticSearchJson.convertToJson(spamMail))
+        assertThatJson(messageToElasticSearchJson.convertToJson(spamMail, ImmutableList.of()))
```
Comment (mbaechler) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/MailboxMessageToElasticSearchJsonTest.java]: what is this empty list ?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java [hunk 5]
```diff
@@ -203,47 +195,47 @@ protected boolean matches(SearchQuery.TextCriterion criterion, MailboxMessage me
...
 
-    protected boolean isInMessage(String value, InputStream input, boolean header, Logger log) throws IOException, MimeException {
+    protected boolean isInMessage(String value, InputStream input, boolean header) throws IOException, MimeException {
         MessageMatcher.MessageMatcherBuilder builder = MessageMatcher.builder()
             .searchContents(Lists.<CharSequence>newArrayList(value))
             .caseInsensitive(true)
             .includeHeaders(header);
-        if (log != null) {
-            builder.logger(log);
+        if (session != null) {
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java]: can getLog() return null ?

Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java]: you don't check for nullity at other call sites

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java [hunk 6]
```diff
@@ -114,6 +114,9 @@ private static UidCriterion findConjugatedUidCriterion(List<Criterion> crits) {
     }
     
     private Multimap<MailboxId, Long> searchMultimap(MailboxSession session, Mailbox mailbox, SearchQuery query) throws MailboxException {
+        if (!mailbox.getUser().equals(session.getUser().getUserName())) {
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java]: why ? would you write that in a way that express your intent ?

File: mailbox/store/src/test/resources/eml/htmlMail.json [hunk 7]
```diff
@@ -158,5 +158,7 @@
   "isFlagged":false,
   "isRecent":false,
   "hasAttachment":true,
-  "isUnread":false
+  "isUnread":false,
+  "users": [
```
Comment (mbaechler) [inline mailbox/store/src/test/resources/eml/htmlMail.json]: they will never match anything, right ?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java [hunk 8]
```diff
@@ -139,15 +142,9 @@ private static UidCriterion findConjugatedUidCriterion(List<Criterion> crits) {
         }
         
         // MessageSearches does the filtering for us
-        if (session == null) {
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java]: I don't see any test that covers this case

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java (hunk 1 above)
Comment (aduprat) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java]: OK

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/query/QueryConverter.java (hunk 2 above)
Comment (aduprat) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/query/QueryConverter.java]: OK

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMailboxMessageSearchIndexTest.java (hunk 3 above)
Comment (aduprat) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMailboxMessageSearchIndexTest.java]: OK

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/MailboxMessageToElasticSearchJsonTest.java (hunk 4 above)
Comment (aduprat) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/MailboxMessageToElasticSearchJsonTest.java]: OK

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java (hunk 5 above)
Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java]: OK

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java (hunk 6 above)
Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java]: OK

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java (hunk 8 above)
Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndex.java]: OK

File: mailbox/store/src/test/resources/eml/htmlMail.json (hunk 7 above)
Comment (aduprat) [inline mailbox/store/src/test/resources/eml/htmlMail.json]: Done

**Comment by aduprat** [comment]
#348

**Pull request closed without merging**